### ISSUE TITLE
Fix false positives for SCSS variables in function-calc-no-unspaced-operator

### ISSUE THREAD

**Issue by peter-neumann-dev** [stylelint/stylelint#6052]
### What steps are needed to reproduce the bug?

```scss
textarea {
  min-height: calc(#{$line-height} * 5);
}
```

### What Stylelint configuration is needed to reproduce the bug?

```js
module.exports = {
  plugins: [
    'stylelint-scss',
  ],
  extends: ['stylelint-config-standard-scss'],
  rules: { },
}

```

### How did you run Stylelint?

CLI with `stylelint 'css/**/*.scss'`

### Which version of Stylelint are you using?

`14.8.1`

### What did you expect to happen?

Behavior like in previous versions <= 14.8.0

### What actually happened?

```shell
css/components/_forms.scss
 213:27  ✖  Expected single space before "-" operator  function-calc-no-unspaced-operator
 213:28  ✖  Expected single space after "-" operator   function-calc-no-unspaced-operator

```

### Does the bug relate to non-standard syntax?

Yes it happens with SCSS variables inside calc()

### Proposal to fix the bug

This happens only in the newest version 14.8.1 with this fixed: https://github.com/stylelint/stylelint/pull/6045

**Comment by jeddy3** [comment]
@peter-neumann-dev Thanks for the request and for using the template.

I can reproduce with [the demo](https://stylelint.io/demo#N4Igxg9gJgpiBcIAuMAeSCGAnGGAEwAOgHZ54C2AlsQLQAWMlA5nUvHmBgDZgAUAxMAAkXajHqMWSAL54AVHgCsASgDcJaSRAAaEAGcAnsUyoE4PXp3gIxAGbMzRUnkIgsAVy4w9r9k7JkrhhINB5eNMQQNO7EANaRAO7EvnhIHjDaJAEuIABGXBBgsRFRMOQADkgGKWnuGVkBrpAFWCU01ABu3JRQEqg16ZnOgdbk5DDGbWWV1Qipgw0jYO56SBDkNOVYEOUwWFVtVBbUTDRdrbYxYEiUNgN1Q9musGBc2MG3tPmFxZE0UO5yqJOCgaMtVutNttdvtKN57vVhjkXm8sB8bDRvkU2gCgZQQeItjs9jd4XMANqLAK1RHZMj+OkjZiRHApSlIxlNGx6GDLG4dcS44HBbw0BKUJB0f6UWy2PYTEJdLh1HwgKnZAC66rwmiRGsejRAKPeNwxWN+UT0dAg+zoGGIvSJMIOEAFWCwPTJ7BpBpGthsIVsGCoXAMOMBwtBxGDXvmDyprn9kyDIbDfyOehONCYEz2+JoKcooZosRgBgSNqgCN9OUuxGunzB3DAbRienKGDAMF6xLRayw1YTIDrDYxomIuFaTDRUDhkz+kWIq3tUGwvVnOFHyTmPqHI9NtD+MXiECSg6RrlLBlsaPG-15qPRh6ilAqNswxnPTxA41nGALuBIO4OARDGrZxIk27egsF4gNGv7Zh6vTYLgehtJ03RVjuMHfn8sB6F2DpZu2vIyviEqzNB8awXhEb4iKNDBO0b77KEnixruNFREK9GgjyXjXDaqpUbShp-NMBx6BAwFdl+YmlOgaI0DyVDNNyckjH8GGiOu0n5OIehvFaYLrOMxjCXGomaS+xBKj0mwQJmB6MSEr7lO+bFeBpORae6MBMJ42BinQEreB2snYdR35OiSaZRMekHea4-G8v2mw8u4UBRK8GAWOBJ5npFVk5ClgmtOUGVZTQMBeGZIRHhBp5QZZNbJTVqU2jQVS7PliUUtqnGMgynIgMyNowGy2rfuCawbO1dWqlNeBahyOpUvqQ6rB6xCnAuMAJOOE1Fa1IAxBKvVNUlIDcOUdpnNwdQlJgB6TatrgwoRmA5q4Nb0ktrhoF2lQAArQiScIWeyjLfjsnYUa4S0rYyuqaidTFhOIElhodmIwP6rL9W911cAkGAGKqv0EP9ICAzAlSvdDhrml4eUYLYKCtHoYHweIzPeBTS1+pQWCrBE3goFhiOU0yTAskd7BQ4zORsxzJljAqCOrWQSN0ijAQbbB6PsaBd6cDyKSuAUCR7D9Q5G+EPPKeF4gqzbcxBCTZNoZmO3hIdtuGyEGNtAKDqdVseOUP0x120HxsqfiEAFIe+0467A7u8TpPkwH37mmCBQ+6cuRol21UVAcOO5Hj40W3BMBurnTMFNiryOVmJeduIE4HWIjHs277Ae9nAuwfnbdF5ipfd6nffV-j8vK575M0OQng3DQ-s6EO4+Fx309O13uML3X3Aj8pJx+2ITcjPnxLEPvR892nA8Z0PWde6v6+UJv1-b2PLdij30fmXdsR906n2Xt7S+4gt4nTvrsB+O0p5HzAWXeetdM5ny9jfHIakLhXGcpEZ6nw67kGgHsbcJ18F9CbObTOVs3bUKTp1BgqBN4TCYJKOuVp3y4K5OrSYWNf4TmPpghW2ph44P-kNamtN6aEyVrWYWosJyrG7JrRmBtGauDGgTCRWtDSrAMF4ccIRIBjBXItQxOtsh621swwRIQEghRQGg8Q1BMywEgSPfh4AVizShL2A4wiq4130XgRW35sE5xkXSYadIAaoCBkgBmSsgivzBAEyEMV9izGloaewIt6riw0WqGxBTdGy3EZEwWytMkWLqrgkanjPQX19rAuegDNHQ1sQEexy1qHZI2L+Sg-4OxIA5s1KJhoAB6vByT-gAF4akWTQJZAAGGgABODUchlC8BoGszZOyNQAGplByCEM0qmhjXDjAsBgb6mcACiqBdjXG7BwYZFBuxjLwAARzqFgAweAeapAgHgaueBSy5AwLkOhR1tQDO0YaGaOSwZ5M2MEKZaTvzzOOas5ZWzdn7MOcckl5zLnXLidkBJ34Hlc2ee-N5HyJbfIhOQPAuSqhgpjBCqFMAYUwDhQis2SLVoopOsaPsjY4XIPcf3DmdcJyN1pciB8Jo5X2lOIqjBESpGxOlZq2VZpAHKTKInZOYte6iIgVgqBX8uAbzgUOGVT5MTmoTmpWgir7XvxidAjpIijrGpyqar4XrLU+sPug8Ji9XCqqYW6k1Hr85FyvqI8gGB2HuoPBZAAjGGx8zl85pAwEWYi0aWHNUNaPb8ea5Xmr+DgAEDp7QhGTiwFcQTnQQyuo2jEMbn5939Uvc+a9nU-1dbBQdtAY1+tfj4z+GbOkTj8XOky1q9XxpVQ3ZNs7U3OVCXPXdijEkfyNdqelhp5GpPPTo66mS503O-EU1RpTJYVOpnoxeMyRrpzVk09VLSlxtNXSGz1PwemMj6WQKVe4AwFmDEWNMYEgUQDcculeLj5ShF5KZCYsAsInX3I2Rp-4R12qXQ6id38Q1+LI0O0y-5F3KtoyumBDH1WJkIeRljsbeZnvfkmjOpG+MYmzewk9aiUgbPE-WIhYFxV10YWJvcEnaAdhwMYBgPI0JUY8WB7xHGV6TpdX-BTW5sU6clN4UUirWkmZE-uzmXGZ1vs09ELAXAaAYaw6Z+thSvMuNCmxwe47pEnToI9WA04YDd0w0+U+HS-HUFgMYZLcwABMJ0rw3hjAZsCEzcUPrmQs5ZRL1mUrJUc4lpyLlXJuTekYjKnkJpAKy1KXz8u3iFeCtYgrhWisRTB-p60TpeB2pKdZewogNQlFdKT5cZghsLSdZbOMptcLoCkAt2X5NDlGf+OUwRgLiAXc7JVEW61+OOwBM7IFLtH31R10Td2-kncAudk2F3codbUx9v8D2gIgUdn8UOWVyo4HsNHESG3Psg5+9phUemHNXac291zQOxlI5AmiHa4hezBE6uFt+kWr2wXu6d0H4gCc5hoMTtKO6T6BZx-+IFexsaUFFhRm1L92MBsdeZ6dlmjuI85yC3+vOBNk+w0GpgmbQ3i+B5L7nMvLGCbEQa+uaqTriQrmGd9na-7w6HOJJOwUwvOyuumHnk8iOM9sMpaSWAy6Gdt+4cg1dWheAwLOZBSy5vy78cQL3PvV45qhKRTMdw5gABZ9fh72G0ctlbA9zYsoNQ0PKwwqYYaeA90VMUHAh0R8OMOo5XWDjJoTrODEAeF-Rjz8S5HJLpvehv6ThwqJKeor9WjKmjWqRE-9F7AONI1uUrRyKJubXamVFyaRKC5HcKCTuRQYBIG9hj4zWO9fz4EmlHFHpV+giZ6Tq7Y7E3Y54-oBfx-Jmn7X0TmEJPOZXde3ug-sFSqP+X2fuIP5rGLdnfn-p1DlHlCVpQniuVoShSqcrVggbso1jSpTC1jkG1syjkF1p8lABwEZHoHgOAVgAKtCrCvCqNtPrrHPr-g-hAesLkNQO-lrtfpekFiMCQWrEwdGMzp-sJhThwSVPQa0PhIRKuJMBYjwSwQuBiO4ldFwfZNAVgNMpIiAASpVsgXsgcnVtVg1tSs1tTFgR1rgeyj0MQSIWQUKhQWKv9mNvBrQd+FwaIBrtmvzqOjRkLr4mASIdLuYrLvwfXj5LfidFwctrXhZIdnQUfp1GXmHNDpHHDi1Ifh1OVJVNlIQVQe-IDj4TEWkTAJlBkblGhCjrpvZjvkfJjt-kXkYr4RVAUVVPNAqFkTkDkaEXUekdVLVM0cOkli9JnFlO4PpH4lwd1H9vQtkYXupr-taLaD2rnm0K2jEBIYqA9BxDhEYsvsgsARZM8HpF5HfmdOYnYQXtbNMd+EqI9FeBWFgL0PnpMWcX4pcZ0q4ZRrPNRoLoIU6hZuunfs8X4UBqxlfp4V8RBi3iMP8S4f4ZrizjUjfj-hcWsQCeEYbmtnJidP8XEVDtHrDgiA0LqNICANIEAA).

I've labelled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.

It should just be the case of using the [`isStandardSyntaxMathFunction`](https://github.com/stylelint/stylelint/blob/main/lib/utils/isStandardSyntaxMathFunction.js) util on Line 310:

https://github.com/stylelint/stylelint/blob/e017583128863ee75634cb15e721779163b571e5/lib/rules/function-calc-no-unspaced-operator/index.js#L309-L311

And then updating the tests to reflect that the rule will now outright ignore any `calc` functions that contain non-standard syntax.

**Comment by ybiquitous** [comment]
@peter-neumann-dev Thanks for the report and sorry for the inconvenience.
I've opened PR #6053 to fix the regression. I expect the next version will include the fix.
